ios/RCTContacts/RCTContacts.m

```diff
@@ -404,52 +423,92 @@ -(NSString *) getFilePathForThumbnailImage:(NSString *)recordID
     if(!contactStore) {
         contactStore = [[CNContactStore alloc] init];
     }
-    
+
     NSString* recordID = [contactData valueForKey:@"recordID"];
-    
+
     NSArray *keys = @[CNContactIdentifierKey,
                       CNContactEmailAddressesKey,
                       CNContactBirthdayKey,
                       CNContactImageDataKey,
                       CNContactPhoneNumbersKey,
                       [CNContactFormatter descriptorForRequiredKeysForStyle:CNContactFormatterStyleFullName],
                       [CNContactViewController descriptorForRequiredKeys]];
-    
+
     @try {
-        
+
         CNContact *contact = [contactStore unifiedContactWithIdentifier:recordID keysToFetch:keys error:nil];
-        CNContactViewController *controller = [CNContactViewController viewControllerForContact:contact];
-        
-        controller.delegate = self;
-        
+        CNContactViewController *contactViewController = [CNContactViewController viewControllerForContact:contact];
+
+        // Add a cancel button which will close the view
+        // TODO localize cancel button title (either through creating a localized strings file, or passing in the title)
+        contactViewController.navigationItem.backBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Cancel" style:UIBarButtonItemStylePlain target:self action:@selector(cancelContactForm)];
+        contactViewController.delegate = self;
+
+
         dispatch_async(dispatch_get_main_queue(), ^{
-            // TODO: Change so this is not opened as a modal. We want the back button!
-            UINavigationController* navigation = [[UINavigationController alloc] initWithRootViewController:controller];
-            UIViewController *viewController = (UIViewController*)[[[[UIApplication sharedApplication] delegate] window] rootViewController];
-            [viewController presentViewController:navigation animated:YES completion:nil];
-            
+            UINavigationController* navigation = [[UINavigationController alloc] initWithRootViewController:contactViewController];
```

Comment:
I have tested it and I can see one small thing that we probably can ignore. If I press cancel the Done button in the top right corner flicks to Edit before the view disappears.

Reply:
Ok, yes I think we will ignore it.